utils/benchmark-operator.sh

```diff
@@ -52,8 +52,15 @@ remove_benchmark_operator() {
 ############################################################################
 run_benchmark() {
   source ${ripsaw_tmp}/bin/activate
-  set -e
-  ripsaw benchmark run -f ${1} -t ${2}
-  set +e
+  if ! ripsaw benchmark run -f ${1} -t ${2}; then
+    log "Benchmark failed, dumping workload more recent logs to stdout"
+    kubectl -n benchmark-operator get pod -l benchmark-uuid=${UUID}
+    for pod in $(kubectl -n benchmark-operator get pod -l benchmark-uuid=${UUID} -o custom-columns="name:.metadata.name" --no-headers); do
+      log "Pod ${pod}"
+      kubectl logs --prefix --since=30s ${pod}
```

Comment:
Instead of just dumping what could be a good amount of logs to stdout what about a file named after each pod into a errorLog dir? that way its easier to grep through later and its not lost when the session is gone.

Reply:
IIRC, benchmarks are executed within an airlflow worker pod, which is ephemeral :cry

Reply:
its not ephemeral when i run it outside of airflow :-D

Reply:
I don't follow - kubectl logs is called outside the pod, so it'd write the log file to the host, right?

Reply:
how about an option for either? just worried about loss to my tmux buffer lol

Reply:
@mkarg75 airflow is running e2e from w/in a pod. When the test is done that pod goes poof. So storing the file locally there doesnt help.

Reply:
ah, I was missing the link to airflow. thanks

Reply:
We can just print the latest 20 lines rather than the latest 30 secs. That should work fine for every terminal buffer

Reply:
@dry923, I added a new commit to dump pod logs to files in addition to the 30 latest lines to stdout

Reply:
sounds good @rsevilla87 . we can roll w/ that for now and adjust later if needed. :+1: